woocommerce/woocommerce-gateway-stripe
# Account Keys Connection Status

Fixes #1806

## Changes proposed in this Pull Request:







This PR adds the ability to test whether a **Publishable Key** and **Secret Key** combination are valid. It does this by adding a "Test connection" link in the **Edit account keys & webhooks** modal (see image below). 

A tradeoff that we should be aware of, is that "test" tokens are created in the merchant's Stripe account when the connection is tested. These tokens do not contain any useful data but instead are simply used to test the validity of the publishable and secret keys.

![Screen Shot 2021-09-08 at 4 12 00 PM](https://user-images.githubusercontent.com/273592/132586292-52e6a387-b51e-43a2-80a1-5f4ddde61439.png)

## Testing instructions





- As a merchant, go to `WooCommerce > Settings > Payments > Stripe` and click on the `Settings` tab (at the top).
- Click on the `Edit account keys` button in order to open the **Edit test account keys & webhooks** modal.
- Enter publishable and secret keys and click on the `Test connection` link.
- Ensure publishable and secret keys result with a **Connection successful!** response, if they are valid. **We couldn't connect.**, otherwise.


---


**Post merge**
